pvnet/models/multimodal/site_encoders/encoders.py

```diff
@@ -244,3 +244,295 @@ def forward(self, x):
         x_out = attn_output.squeeze()
 
         return x_out
+
+
+class SingleWindAttentionNetwork(AbstractPVSitesEncoder):
```

Comment:
It seems like there is probably quite a lot of duplication here. Could we do some inheritance to clean things up?